sql/catalyst/src/main/java/org/apache/spark/sql/connector/metric/MergeMetrics.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connector.metric;
+
+import java.util.OptionalLong;
+
+/**
+ * Execution metrics for a Merge Operation for a Connector that supports RowLevelOperations
```

Comment:
IIRC different MERGE strategy have different metrics, do we need to separate them?

Reply:
do you mean MERGE actions?  Actually should always be same metrics (some will be 0 if not activated).  Anyway , removed this class in favor of more generic Map.